### PR TITLE
Update retry dep to match mismi.

### PR DESCRIPTION
**Pull request by tmcgilchrist**
@charleso

### CODE REVIEW

**Comment by charleso** [comment]
I would be tempted to put a version range in, otherwise you'll have to upgrade this with mismi in lock-step.

**Comment by markhibberd** [comment]
> I would be tempted to put a version range in, otherwise you'll have to upgrade this with mismi in lock-step. 

Are they compatible? I thought the problem was they weren't.

**Comment by tmcgilchrist** [comment]
They're not compatible, the type is now an alias for `RetryPolicy m`  `type RetryPolicy = forall m. Monad m => RetryPolicyM m`

**Comment by tmcgilchrist** [comment]
@charleso not entirely convinced I know why this is working now.

File: src/Snooze/Balance/Core.hs [hunk 1]
```diff
@@ -82,17 +83,25 @@ randomRoundRobin' policy f l =
...
     r <- lift $ f e
     put $ case r of
       Left l -> (Nothing, l : e)
       Right b -> (Just b, e)
     return r
+-- retrying' policy f =
+--   flip execStateT (Nothing, []) . retrying policy (\_ -> return . isLeft) $ do
+--     (_, e) <- get
+--     r <- lift $ f e
+--     put $ case r of
```
Comment (charleso) [inline src/Snooze/Balance/Core.hs]: Cruft?

File: test/Test/Snooze/Server.hs [hunk 2]
```diff
@@ -26,10 +26,11 @@ withServer app f = do
   port <- randomRIO (10100, 65534)
   -- Check that we can connect first to avoid flakey "connection refused"
   let connect' = bracket (connectTo "localhost" $ PortNumber (fromInteger $ toInteger port)) hClose pure
+  let def = limitRetries 5
   bracket
     (forkIO . scotty port $ app)
     killThread
-    (const $ (recoverAll (limitRetries 5) connect') >> f (requestCreate "localhost" port))
+    (const $ (recoverAll def (\_ -> connect')) >> f (requestCreate "localhost" port))
```
Comment (charleso) [inline test/Test/Snooze/Server.hs]: `def`? Sigh :(

File: src/Snooze/Balance/Core.hs [hunk 3]
```diff
@@ -82,14 +83,14 @@ randomRoundRobin' policy f l =
...
+   RetryPolicy -> ([e] -> a -> m (Either e b)) -> [a] -> m (Maybe b)
 randomRoundRobin policy f =
   fmap fst . randomRoundRobin' policy f
 
 -- FIX Should be lifted in to x-retry (or thereabouts)
-retrying' :: MonadIO m => RetryPolicy -> ([e] -> m (Either e b)) -> m (Maybe b, [e])
+retrying' :: (MonadIO m) => RetryPolicy -> ([e] -> m (Either e b)) -> m (Maybe b, [e])
 retrying' policy f =
-  flip execStateT (Nothing, []) . retrying policy (\_ -> return . isLeft) $ do
+  flip execStateT (Nothing, []) . retrying policy (\_s e -> return . isLeft $ e) $ \_ -> do
```
Comment (charleso) [inline src/Snooze/Balance/Core.hs]: Unused `s`?

**Comment by charleso** [comment]
Seems fine :+1:

File: test/Test/Snooze/Server.hs (hunk 2 above)
Comment (tmcgilchrist) [inline test/Test/Snooze/Server.hs]: I'll inline that, was just trying to tease apart the types when I did that.

**Pull request merged**